ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -144,6 +163,24 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
             </>
         );
     }
+
+    private static getCollapsedNodeName(parent: string, hiding: number): string {
+        return '@collapsed/' + parent + '/' + hiding;
+    }
+
+    private static isCollapsedNode(id: string): boolean {
+        return id.startsWith('@collapsed/');
+    }
+
+    private static getCollapsedNodeParent(id: string): string {
+        const split = id.split('/');
+        return split[1];
+    }
+
+    private static getCollapsedNodeHiding(id: string): number {
+        const split = id.split('/');
+        return +split[2];
```

Comment:
why pre-increment?

Reply:
This is not a pre-increment, it's a cast from `string` to `number`

Reply:
eh? can you use `as number` instead then?

Reply:
Made it `Number(split[2])` to be clearer